bin/glow-monitor/main.go

```diff
@@ -4,23 +4,44 @@ import (
 	"fmt"
 	"os"
 	"os/signal"
+	"path/filepath"
 	"syscall"
 
 	"github.com/glowlabs-org/gca-backend/client"
 )
 
 func main() {
 	// Create a new client, using the current directory as the basedir.
-	c, err := client.NewClient("/opt/glow-monitor/")
+	baseDir := "/opt/glow-monitor/"
+	c, err := client.NewClient(baseDir)
 	if err != nil {
 		fmt.Println("unable to create client: ", err)
 		return
 	}
 
 	// Wait for a shutdown signal from the OS.
 	sigChan := make(chan os.Signal, 1)
-	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
-	<-sigChan
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGUSR1, syscall.SIGUSR2)
+
+	done := false
+	for !done {
+		s := <-sigChan
+		switch s {
+		case syscall.SIGINT:
+			done = true
+		case syscall.SIGTERM:
+			done = true
+		case syscall.SIGUSR1:
+			// Dump status to the terminal.
+			fmt.Printf("%v", c.DumpStatus())
+		case syscall.SIGUSR2:
+			// Write the status to a file "status.txt" in the client directory. File will be
+			// created and/or truncated before writing.
+			path := filepath.Join(baseDir, "status.txt")
+			fmt.Printf("Dumping server status to %v\n", path)
+			os.WriteFile(path, []byte(c.DumpStatus()), 0644)
+		}
+	}
```

Comment:
have you tested this? Have you tested what happens if you send a signal that isn't handled by the switch statement?

Reply:
I have tested everything from command line. If you send a signal not handled by the switch statement, the kernel does whatever the default behavior is here: https://www.man7.org/linux/man-pages/man7/signal.7.html

For example, sending kill -usr1 without this switch statement ignores. sending kill -hup terminates the process, etc.

Reply:
The line `signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGUSR1, syscall.SIGUSR2)` ensures that only the signals specified in the switch statement will be sent to the channel, others will be handled by the kernel